src/encoder.rs

```diff
@@ -1251,6 +1248,16 @@ pub fn encode_tx_block<T: Pixel>(
       visible_tx_h,
     );
   }
+  unsafe {
```

Comment:
init_slice_repeat_mut doesn't do the same but using safe code?

Reply:
init_slice_repeat_mut works on a slice of MaybeUninit, but I can do this with a for loop (which is what init_slice_repeat_mut does internally) and it compiles down to a memset.

Interestingly, using `slice.iter_mut.for_each()` does not compile down to a memset. but using `for a in slice.iter_mut()` does. Weird.

Reply:
I mean using the MaybeUninit array + unit_slice_repeat_mut so we get the correct slice. This version is already nice but if it works the same I'd keep the code more uniform.

Reply:
I think the complexity around using init_slice_repeat_mut here is that we initialize part of the array with real values while running the `diff` function. At which point we already need a `&[i16]`. So it's simpler, in my opinion, to not use MaybeUninit in this case.